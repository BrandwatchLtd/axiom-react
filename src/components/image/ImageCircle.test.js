import React from 'react';
import renderer from 'react-test-renderer';
import { ImageCircle } from 'bw-axiom';

function getComponent(props = {}) {
  return renderer.create(
    <ImageCircle { ...props } size="3rem" />
  );
}

function matchSnapshot(props = {}) {
  const component = getComponent(props);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
}

describe('ImageCircle', () => {
  it('renders with defaultProps', () => {
    matchSnapshot();
  });

  it('renders with border small', () => {
    matchSnapshot({ border: 'small' });
  });

  it('renders with border large', () => {
    matchSnapshot({ border: 'large' });
  });

  it('renders with overlay icon', () => {
    matchSnapshot({
      overlayIconName: 'twitter',
      overlayIconSize: '2rem',
    });
  });
});
