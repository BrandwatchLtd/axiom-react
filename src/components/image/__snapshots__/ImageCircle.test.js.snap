exports[`ImageCircle renders with border large 1`] = `
<div
  className="ax-image-circle ax-image-circle--border-large"
  style={
    Object {
      "height": "3rem",
      "width": "3rem",
    }
  }>
  <div
    className="ax-image-circle__image">
    <img
      className="ax-image"
      onError={[Function]}
      style={
        Object {
          "height": "3rem",
          "width": "3rem",
        }
      } />
  </div>
</div>
`;

exports[`ImageCircle renders with border small 1`] = `
<div
  className="ax-image-circle ax-image-circle--border-small"
  style={
    Object {
      "height": "3rem",
      "width": "3rem",
    }
  }>
  <div
    className="ax-image-circle__image">
    <img
      className="ax-image"
      onError={[Function]}
      style={
        Object {
          "height": "3rem",
          "width": "3rem",
        }
      } />
  </div>
</div>
`;

exports[`ImageCircle renders with defaultProps 1`] = `
<div
  className="ax-image-circle"
  style={
    Object {
      "height": "3rem",
      "width": "3rem",
    }
  }>
  <div
    className="ax-image-circle__image">
    <img
      className="ax-image"
      onError={[Function]}
      style={
        Object {
          "height": "3rem",
          "width": "3rem",
        }
      } />
  </div>
</div>
`;

exports[`ImageCircle renders with overlay icon 1`] = `
<div
  className="ax-image-circle"
  style={
    Object {
      "height": "3rem",
      "width": "3rem",
    }
  }>
  <div
    className="ax-image-circle__image">
    <img
      className="ax-image"
      onError={[Function]}
      style={
        Object {
          "height": "3rem",
          "width": "3rem",
        }
      } />
  </div>
  <div
    className="ax-image-circle__icon">
    <svg
      className="ax-icon"
      style={
        Object {
          "height": "2rem",
          "width": "2rem",
        }
      }
      viewBox="0 0 16 16">
      <path
        d="M1,12.087 C1.225,12.113 1.454,12.127 1.685,12.127 C3.031,12.127 4.27,11.667 5.252,10.897 C3.995,10.874 2.935,10.044 2.569,8.902 C2.745,8.936 2.925,8.954 3.11,8.954 C3.372,8.954 3.716,8.919 3.957,8.853 C2.643,8.589 1.743,7.429 1.743,6.037 C1.743,6.216 2.483,6.345 2.954,6.36 C2.183,5.845 1.631,4.966 1.631,3.969 C1.631,3.443 1.75,2.949 1.997,2.525 C3.414,4.263 5.519,5.407 7.906,5.526 C7.857,5.316 7.826,5.097 7.826,4.872 C7.826,3.286 9.11,2 10.696,2 C11.522,2 12.267,2.348 12.791,2.907 C13.445,2.778 14.059,2.539 14.614,2.209 C14.399,2.88 13.944,3.443 13.351,3.799 C13.932,3.729 14.485,3.575 15,3.346 C14.615,3.922 14.128,4.428 13.567,4.833 C13.573,4.956 13.575,5.08 13.575,5.205 C13.575,9 10.686,13.377 5.403,13.377 C3.781,13.377 2.271,12.901 1,12.087 Z" />
    </svg>
  </div>
</div>
`;
